--- /tmp/tmp.aoqOpC7GAY/lua/pac3/core/server/init.lua	2026-05-06 23:34:10.000000000 +0000
+++ /home/cookie/gmod/serverfiles/garrysmod/addons/0020_pac3-master/lua/pac3/core/server/init.lua	2026-08-18 16:45:35.797552062 +0000
@@ -22,6 +22,10 @@
 
 include("pac3/core/shared/init.lua")
 
+-- change by cookie9216
+include("security_validate.lua")
+include("security_policy.lua")
+
 include("effects.lua")
 include("event.lua")
 include("net_messages.lua")
